lib/rspec/core/option_parser.rb

```diff
@@ -68,7 +68,17 @@ def parser(options)
 
         parser.on('--bisect[=verbose]', 'Repeatedly runs the suite in order to isolate the failures to the ',
                   '  smallest reproducible case.') do |argument|
-          bisect_and_exit(argument)
+          options[:runner] = -> (opts, err, out) do
+            RSpec::Support.require_rspec_core "bisect/coordinator"
+
+            success = Bisect::Coordinator.bisect_with(
+              opts.command_line_and_env_args,
+              RSpec.configuration,
+              bisect_formatter_for(argument)
+            )
+
+            exit(success ? 0 : 1)
+          end
```

Comment:
I think it's a bit easier to read if all the runner actions have a method that implements it, so that the parsing logic here can just be `options[:runner] = method(:bisect_and_exit)`.